cluster-autoscaler/config/autoscaling_options.go

```diff
@@ -125,7 +126,11 @@ type AutoscalingOptions struct {
 	IgnoreMirrorPodsUtilization bool
 	// MaxGracefulTerminationSec is maximum number of seconds scale down waits for pods to terminate before
 	// removing the node from cloud provider.
+	// MaxGracefulTerminationSec will not be applicable when the scale-down-drain-node-priority-and-grace-periods flag is set
```

Comment:
The comments here should reference field names, not flag names.

Currently, `MaxGracefulTerminationSec` is always set, and additionally `DrainPriorityConfig` is set if `--scale-down-drain-node-priority-and-grace-periods` is passed.  So in the latter scenario, `MaxGracefulTerminationSec` is set to the default value of the `--max-graceful-termination-sec` flag, which is very misleading as this value is not actually used for anything.

Can we reverse this, and always have `DrainPriorityConfig` set (constructed from either of the 2 flags), while only setting `MaxGracefulTerminationSec` if `--max-graceful-termination-sec` is passed? We can still detect that the old flag was used in Actuator constructor by doing `ctx.MaxGracefulTerminationSec > 0`. This IMO makes for much clearer semantics - `DrainPriorityConfig` fully controls drain behavior as the only setting, while `MaxGracefulTerminationSec` only tells you whether `DrainPriorityConfig` was constructed from the legacy flag (and it's value either matches `DrainPriorityConfig`, or is 0 - less chance of someone assuming it matters).

Reply:
Comments are updated.

Discussed offline and decided to proceed with the current implementation.